src/DynamoCore/Graph/Nodes/NodeModel.cs

```diff
@@ -1050,23 +1169,6 @@ public bool IsTopMostNode
             }
         }
 
-        internal void ConnectInput(int inputData, int outputData, NodeModel node)
```

Comment:
@ikeough @ramramps and I talked about this in more depth, and I think if you want to make this change it needs to be profiled on a graph with 1000 + nodes, before and after the change.

We believe this may be very slow  when freezing a node - what will happen is that `ComputeUpstreamCache ` and `ComputeUpstreamOnDownstreamNodes` will be called on every node, or large subgraphs of nodes when properties are changed, like freeze which relies on parents state. These methods rely on the precomputed inputs - it will at least introduce a constant factor increase per node, as the inputs will be recalculated as the graph is traversed.

I don't think the performance impact will be as bad as the issue we mentioned originally, but these caches are indeed tied together.